### DISCUSSION TITLE
about using `system-upgrade-controller` upgrade k3s

### DISCUSSION THREAD

**Discussion by wang-xiaowu** [rancher/system-upgrade-controller#289] [Q&A] [answered]
i'm preparing using `system-upgrade-controller` upgrade k3s 1.24.16 to 1.25.16，and i still have one or more PodSecurityPolicy resource(s) in this cluster，so i am wondering can i upgrade directly, or i need to handle the PodSecurityPolicy manually

**Comment by brandond** [comment, marked as answer]
You would want to remove PSPs before upgrading. The upgrade itself will work fine, but you may find it difficult to manage charts or other things after you've upgraded, if the current chart deployments refer to resources that are no longer available.
